lightning-net-tokio/src/lib.rs

```diff
@@ -226,7 +226,7 @@ pub fn setup_inbound<CMH, RMH, L, UMH>(peer_manager: Arc<peer_handler::PeerManag
 	#[cfg(debug_assertions)]
 	let last_us = Arc::clone(&us);
 
-	let handle_opt = if let Ok(_) = peer_manager.new_inbound_connection(SocketDescriptor::new(us.clone())) {
+	let handle_opt = if let Ok(_) = peer_manager.new_inbound_connection(SocketDescriptor::new(us.clone()), None) {
```

Comment:
We should be able to fill this and the `new_outbound_connection` ip addresses using `TcpStream.peer_addr`.

Reply:
`TcpStream::peer_addr` is not of the same type as required by the `new_outbound_connection`(NetAddress).
So I had to convert it into a `NetAddress` type. This is what I came up with:
```
	let ip_bytes = match TcpStream::peer_addr().unwrap().ip() {
		IpAddr::V4(ip) => ip.octets().to_vec(),
		IpAddr::V6(ip) => ip.octets().to_vec(),
	};
	let handle_opt = if let Ok(initial_send) = peer_manager.new_outbound_connection(their_node_id, SocketDescriptor::new(us.clone()), match TcpStream::peer_addr().unwrap().ip() {
		IpAddr::V4(ip) => Some(NetAddress::IPv4 {
			addr: ip_bytes.try_into().unwrap(),
			port: TcpStream::peer_addr().unwrap().port(),
		}),
		IpAddr::V6(ip) => Some(NetAddress::IPv6 {
			addr: ip_bytes.try_into().unwrap(),
			port: TcpStream::peer_addr().unwrap().port(),
		}),
	}) {
```

But here I am encountering errors of this type
```
278 |             port: TcpStream::peer_addr().unwrap().port(),
    |                   ^^^^^^^^^^^^^^^^^^^^-- supplied 0 arguments
    |                   |
    |                   expected 1 argument
```
I am not sure what argument I am supposed to pass here..

Reply:
`peer_addr` takes `&self`, so its an instance method not a static one.